src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -3735,6 +3772,48 @@ class CgroupUtils(object):
                 raise
         return self.assigned_resources is not None
 
+    def add_jobid_to_cgroup_jobs(self, jobid):
+        pbs.logmsg(pbs.EVENT_DEBUG4, 'Adding jobid %s to cgroup_jobs' % jobid)
+        with open(self.cgroup_jobs_file, 'a+') as f:
+            f.write(jobid + '\n')
+        return True
+
+    def remove_jobid_from_cgroup_jobs(self, jobid):
+        if os.path.isfile(self.cgroup_jobs_file):
+            with open(self.cgroup_jobs_file, 'r+') as f:
+                jobid_list = f.readlines()
+                f.seek(0)
+                for j in jobid_list:
+                    if jobid != j.strip():
+                        f.write(j)
+                f.truncate()
+            pbs.logmsg(pbs.EVENT_DEBUG4,
+                       'Removing jobid %s from cgroup_jobs' % jobid)
+            return True
+        return False
+
+    def read_cgroup_jobs(self):
+        jobids = []
+        if os.path.isfile(self.cgroup_jobs_file):
+            with open(self.cgroup_jobs_file, 'r') as f:
+                jobids = f.readlines()
+            jobids = [line.strip() for line in jobids]
+        return jobids
+
+    def delete_cgroup_jobs_file(self, jobid):
+        pbs.logmsg(pbs.EVENT_DEBUG4,
+                   'Deleting file: %s' % self.cgroup_jobs_file)
+        if os.path.isfile(self.cgroup_jobs_file):
+            os.remove(self.cgroup_jobs_file)
+        return True
+
+    def empty_cgroup_jobs_file(self):
+        pbs.logmsg(pbs.EVENT_DEBUG4,
+                   'Emptying file: %s' % self.cgroup_jobs_file)
+        with open(self.cgroup_jobs_file, 'w+'):
+            pass
```

Comment:
You don't need 'w+', just use 'w'. You need to call truncate() on the descriptor.

Reply:
I put a '+' because '+' will create the file if it does not exist. Or is there a better way to do this?

Reply:
I see. In that case, this is fine.